src/benchq/resource_estimation/openfermion_re.py

```diff
@@ -1,69 +1,92 @@
 ################################################################################
 # © Copyright 2022 Zapata Computing Inc.
 ################################################################################
-from typing import Any, Dict
+from typing import Any, Dict, Tuple
 
 import numpy as np
-import pyscf
 from openfermion.resource_estimates import sf
 from openfermion.resource_estimates.surface_code_compilation.physical_costing import (
     cost_estimator,
 )
 
+from benchq.resource_estimation._compute_lambda_sf import compute_lambda
 
-def model_toffoli_and_qubit_cost_from_single_factorized_mean_field_object(
-    mean_field_object, rank, DE, CHI
-):
-    num_orb = len(mean_field_object.mo_coeff)
+
+def get_single_factorized_qpe_toffoli_and_qubit_cost(
+    h1: np.ndarray,
+    eri: np.ndarray,
+    rank: int,
+    allowable_phase_estimation_error: float,
+    bits_precision_state_prep: float,
+) -> Tuple[int, int]:
+    """Get the number of Toffoli gates and logical qubits for single factorized QPE.
+
+    See get_single_factorized_qpe_resource_estimate for descriptions of arguments.
+
+    Returns:
+        The number of Toffoli gates and logical qubits.
+    """
+    num_orb = h1.shape[0]
     num_spinorb = num_orb * 2
 
     # First, up: lambda and CCSD(T)
-    eri_rr, LR = sf.factorize(mean_field_object._eri, rank)
-    lam = sf.compute_lambda(mean_field_object, LR)
+    eri_rr, LR = sf.factorize(eri, rank)
+    lam = compute_lambda(h1, eri_rr, LR)
 
     # now do costing
-    stps1 = sf.compute_cost(num_spinorb, lam, DE, L=rank, chi=CHI, stps=20000)[0]
+    stps1 = sf.compute_cost(
+        num_spinorb,
+        lam,
+        allowable_phase_estimation_error,
+        L=rank,
+        chi=bits_precision_state_prep,
+        stps=20000,
+    )[0]
 
     _, sf_total_toffoli_cost, sf_logical_qubits = sf.compute_cost(
-        num_spinorb, lam, DE, L=rank, chi=CHI, stps=stps1
+        num_spinorb,
+        lam,
+        allowable_phase_estimation_error,
+        L=rank,
+        chi=bits_precision_state_prep,
+        stps=stps1,
     )
     return sf_total_toffoli_cost, sf_logical_qubits
 
 
-def _get_unsymmetrized_eri(meanfield_object: pyscf.scf.hf.SCF) -> np.ndarray:
```

Comment:
I think it makes sense to keep this as a utility function - would you agree @max-radin?